botorch/acquisition/knowledge_gradient.py

```diff
@@ -46,12 +47,456 @@
 from botorch.exceptions.errors import UnsupportedError
 from botorch.models.model import Model
 from botorch.sampling.samplers import MCSampler, SobolQMCNormalSampler
+from botorch.utils.sampling import draw_sobol_samples
 from botorch.utils.transforms import (
     concatenate_pending_points,
     match_batch_shape,
     t_batch_mode_transform,
 )
 from torch import Tensor
+from torch.distributions import Normal
+
+
+class HybridKnowledgeGradient(MCAcquisitionFunction):
```

Comment:
- make MC the parent class
  - ensure `def sample_z` is a member function that can be overwritten
  - OR make z_vals an argument to the `self.compute_mc` function?
- make hybrid the child class

![image](https://user-images.githubusercontent.com/28802700/149822504-c1092084-af34-4548-928e-10afe434820e.png)